File: frontend/src/types/history.ts
export interface HistoryItem {
  request_id: string;
  created_at: string;
  status: string;
  migration_scope: string;
  source_version: string | null;
  target_version: string | null;
  database_size_gb: number | null;
  recommended_approach: string | null;
  confidence: string | null;
  score: number | null;
  rules_version: string | null;
  recommendation_generated_at: string | null;
}

export interface HistoryListResponse {
  items: HistoryItem[];
  total: number;
}

File: backend/app/schemas/history.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class HistoryItem(BaseModel):
    request_id: str
    created_at: datetime
    status: str
    migration_scope: str
    source_version: str | None = None
    target_version: str | None = None
    database_size_gb: float | None = None
    recommended_approach: str | None = None
    confidence: str | None = None
    score: int | None = None
    rules_version: str | None = None
    recommendation_generated_at: datetime | None = None


class HistoryListResponse(BaseModel):
    items: list[HistoryItem]
    total: int
